pkg/volume/cinder/cinder.go

```diff
@@ -188,15 +192,21 @@ func (plugin *cinderPlugin) newProvisionerInternal(options volume.VolumeOptions,
 	}, nil
 }
 
-func getCloudProvider(cloudProvider cloudprovider.Interface) (CinderProvider, error) {
-	if cloud, ok := cloudProvider.(*openstack.OpenStack); ok && cloud != nil {
-		return cloud, nil
-	}
-	return nil, fmt.Errorf("wrong cloud type")
-}
-
 func (plugin *cinderPlugin) getCloudProvider() (CinderProvider, error) {
 	cloud := plugin.host.GetCloudProvider()
+	if cloud == nil {
+		if _, err := os.Stat(DefaultCloudConfigPath); err == nil {
+			var config *os.File
+			config, err = os.Open(DefaultCloudConfigPath)
+			if err != nil {
+				glog.Errorf("unable to load openstack cloud provider from %s: %v", DefaultCloudConfigPath, err)
+			} else {
+				defer config.Close()
+				cloud, err = cloudprovider.GetCloudProvider(openstack.ProviderName, config)
+			}
+		}
+	}
+
 	if cloud == nil {
```

Comment:
nit: can we move this to the first `if cloud == nil` block

Reply:
@NickrenREN i reworked the logs and moved this back like you said. PTAL. thanks!